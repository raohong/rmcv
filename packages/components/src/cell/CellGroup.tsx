import { isNil } from '@rmc-vant/utils';
import classNames from 'classnames';
import React, { useMemo } from 'react';
import { useConfigContext } from '../config-provider';
import CellContext from './CellContext';
import type { CellContextState, CellGroupProps } from './interface';

const CellGroup = React.forwardRef<HTMLDivElement, CellGroupProps>(
  ({ title, className, children, inset, border = true, size, ...rest }, ref) => {
    const { getPrefixCls } = useConfigContext();
    const baseCls = getPrefixCls('cell-group');

    const ctxValue: CellContextState = useMemo(
      () => ({
        border,
        size,
      }),
      [border, size],
    );

    return (
      <CellContext.Provider value={ctxValue}>
        <div ref={ref} className={className} {...rest}>
          {!isNil(title) && <div className={`${baseCls}-title`}>{title}</div>}
          <div
            className={classNames(baseCls, {
              [`${baseCls}-inset`]: inset,
              [`${baseCls}-border`]: border && !inset,
            })}
          >
            {children}
          </div>
        </div>
      </CellContext.Provider>
    );
  },
);

export default CellGroup;
