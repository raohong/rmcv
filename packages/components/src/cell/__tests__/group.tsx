import React from 'react';
import { render, screen } from '@testing-library/react';
import CellGroup from '../CellGroup';

const testId = 'cell-group';

test('render correctly', () => {
  const tree = render(<CellGroup />);

  expect(tree.asFragment()).toMatchSnapshot();
});

test('render with gutter', () => {
  render(<CellGroup data-testid={testId} title="title" />);

  expect(screen.getByText('title')).not.toBeNull();
});
