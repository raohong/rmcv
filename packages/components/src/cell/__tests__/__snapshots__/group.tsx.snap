// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="rmcvant-cell-group"
    />
  </div>
</DocumentFragment>
`;
